install/dracut-install.c

```diff
@@ -1362,27 +1386,22 @@ static int install_firmware(struct kmod_module *mod)
                                 exit(EXIT_FAILURE);
                         }
 
-                        fw = fwpath;
-                        if (stat(fwpath, &sb) != 0) {
-                                r = asprintf(&fwpath_xz, "%s.xz", fwpath);
-                                if (r < 0) {
-                                        log_error("Out of memory!");
-                                        exit(EXIT_FAILURE);
+                        if (strstr(fwpath, "*") != 0 || strstr(fwpath, "?") != 0 || strstr(fwpath, "[") != 0) {
```

Comment:
With the assumption that `*` `?` `[` will never occur in the filename? At least, check that is was not escaped :-)
Also, I would check `value` instead of `fwpath`.

Reply:
Ah, maybe check if `fwpath` exists, if not check for the regex chars, then do the `glob`

Reply:
Implemented, but in reverse (string comparison should be cheaper than a stat)